From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Schumacher <timschumi@gmx.de>
Date: Wed, 29 Jun 2022 03:48:52 +0200
Subject: [PATCH] Include `sys/time.h` in various places

Apparently, certain definitions from `sys/time.h` are supposed to be
available from `sys/types.h` as well, but there isn't anything to verify
that, so just add the inclusions manually for now.
---
 lib/ext2fs/bitmaps.c     | 1 +
 lib/ext2fs/blkmap64_ba.c | 1 +
 lib/ext2fs/blkmap64_rb.c | 1 +
 3 files changed, 3 insertions(+)

diff --git a/lib/ext2fs/bitmaps.c b/lib/ext2fs/bitmaps.c
index 834a396..6a05b1b 100644
--- a/lib/ext2fs/bitmaps.c
+++ b/lib/ext2fs/bitmaps.c
@@ -21,6 +21,7 @@
 #if HAVE_SYS_STAT_H
 #include <sys/stat.h>
 #endif
+#include <sys/time.h>
 #if HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
diff --git a/lib/ext2fs/blkmap64_ba.c b/lib/ext2fs/blkmap64_ba.c
index 5d8f154..63e6008 100644
--- a/lib/ext2fs/blkmap64_ba.c
+++ b/lib/ext2fs/blkmap64_ba.c
@@ -20,6 +20,7 @@
 #if HAVE_SYS_STAT_H
 #include <sys/stat.h>
 #endif
+#include <sys/time.h>
 #if HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
diff --git a/lib/ext2fs/blkmap64_rb.c b/lib/ext2fs/blkmap64_rb.c
index 0df58dc..57d0346 100644
--- a/lib/ext2fs/blkmap64_rb.c
+++ b/lib/ext2fs/blkmap64_rb.c
@@ -20,6 +20,7 @@
 #if HAVE_SYS_STAT_H
 #include <sys/stat.h>
 #endif
+#include <sys/time.h>
 #if HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
